cmd/rpcdaemon/commands/daemon.go

```diff
@@ -40,32 +40,34 @@ type EthAPI interface {
 
 // APIImpl is implementation of the EthAPI interface based on remote Db access
 type APIImpl struct {
-	remoteDbAdddress string
-	db               *remote.DB
+	remoteDbAddress string
+	db              *remote.DB
 }
 
-func (api *APIImpl) ensureConnected() error {
+func (api *APIImpl) ensureConnected(ctx context.Context) error {
 	if api.db == nil {
-		conn, err := net.Dial("tcp", api.remoteDbAdddress)
+		conn, err := net.Dial("tcp", api.remoteDbAddress)
 		if err != nil {
 			return err
 		}
-		api.db, err = remote.NewDB(conn, conn, conn)
+
+		api.db, err = remote.NewDB(ctx, conn, conn, conn)
 		if err != nil {
 			return err
 		}
+
 	}
 	return nil
 }
 
 // ConnectAPIImpl connects to the remote DB and returns APIImpl instance
-func ConnectAPIImpl(remoteDbAdddress string) (*APIImpl, error) {
-	return &APIImpl{remoteDbAdddress: remoteDbAdddress}, nil
+func ConnectAPIImpl(remoteDbAddress string) (*APIImpl, error) {
+	return &APIImpl{remoteDbAddress: remoteDbAddress}, nil
 }
 
 // BlockNumber returns the currently highest block number available in the remote db
 func (api *APIImpl) BlockNumber(ctx context.Context) (uint64, error) {
-	if err := api.ensureConnected(); err != nil {
+	if err := api.ensureConnected(context.Background()); err != nil {
```

Comment:
Why do we need a new context here?

Reply:
Dilemma is: 
- from one side we wanna reuse connection between http requests. But http.Request.ctx closing on the end of request.
- from another side, wanna close db on SIGTERM (and I think using context is not very bad idea for it). 
- from third side, wanna keep bolt interface (don't add `ctx` param to .Cursor() or .Next() methods...) 

So, don't know how to fit all 3 requirements in same time